From 4f95865d4ee73221d870de0466014a408e7884f6 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Mon, 20 Jan 2025 17:39:29 +0100
Subject: [PATCH 786/841] feat(stm32mp2): support CPU1 reset for M33 TDCID

Add test on the reboot reason "CPU1 reset" to support cortex-A35 restart
after system reset when cortex-M33 is TDCID.
Add test on the reboot reason "CPU1 system Standby" to support
cortex-A35 independent reset (CPU1) after cortex-A35 low power mode
request (when context is valid) and when cortex-M33 is TDCID.

The flags C1RSTF is not enough to detect a CPU1 reset requested by CPU2
because the ROM code request a CPU1 reset to switch to 64 bits execution
mode.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 704155e644eb96d7da9fbb86647b7a7d492358ba feat(stm32mp2): support CPU1 reset for M33 TDCID]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Ibcfd69718f2e8ce7d2dfc4e246cbf811eb1db766
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/430162
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_private.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_private.c b/plat/st/stm32mp2/stm32mp2_private.c
index 1fadb5a56..0d3a93366 100644
--- a/plat/st/stm32mp2/stm32mp2_private.c
+++ b/plat/st/stm32mp2/stm32mp2_private.c
@@ -647,6 +647,10 @@ bool stm32mp_is_wakeup_from_standby(void)
 		if ((rstsr & RCC_C1BOOTRSTSCLRR_PADRSTF) != 0U) {
 			return false;
 		}
+		/* CPU1=CA35 reset performed by CPU2=M33 */
+		if ((rstsr & RCC_C1BOOTRSTSCLRR_STBYC1RSTF) == 0U) {
+			return false;
+		}
 		if (stm32_pm_context_is_valid()) {
 			standby = 1;
 		} else {
-- 
2.47.2

